packages/api/amplify_api/test/web_socket/web_socket_bloc_test.dart

```diff
@@ -117,6 +159,162 @@ void main() {
       await bloc.done.future;
     });
 
+    test('should reconnect when data turns on/off', () async {
+      final dataCompleter = Completer<String>();
+      final subscribeEvent = SubscribeEvent(
+        subscriptionRequest,
+        () {
+          service!.channel.sink.add(mockDataString);
+        },
+      );
+
+      final bloc = getWebSocketBloc();
+
+      expect(
+        bloc.stream,
+        emitsInOrder(
+          [
+            isA<DisconnectedState>(),
+            isA<ConnectingState>(),
+            isA<ConnectedState>(),
+            isA<ReconnectingState>(),
+            isA<ConnectingState>(),
+            isA<ConnectedState>(),
+          ],
+        ),
+      );
+
+      bloc
+          .subscribe(
+        subscribeEvent,
+      )
+          .listen(
+        expectAsync1((event) {
+          expect(event.data, json.encode(mockSubscriptionData));
+          dataCompleter.complete(event.data);
+        }),
+      );
+
+      await dataCompleter.future;
+
+      fakePlatform.controller.sink.add(ConnectivityResult.none);
+
+      fakePlatform.controller.sink.add(ConnectivityResult.mobile);
+    });
+
+    test('should reconnect after repeated wifi toggling', () async {
+      final blocReady = Completer<void>();
+      final subscribeEvent = SubscribeEvent(
+        subscriptionRequest,
+        blocReady.complete,
+      );
+
+      final bloc = getWebSocketBloc();
+
+      expect(
+        bloc.stream,
+        emitsInOrder(
+          [
+            isA<DisconnectedState>(),
+            isA<ConnectingState>(),
+            isA<ConnectedState>(),
+            isA<ReconnectingState>(),
+            isA<ConnectingState>(),
+            isA<ConnectedState>(),
+          ],
+        ),
+      );
+
+      bloc.subscribe(
+        subscribeEvent,
+      );
+
+      await blocReady.future;
+
+      fakePlatform.controller.sink.add(ConnectivityResult.none);
+      fakePlatform.controller.sink.add(ConnectivityResult.wifi);
+      fakePlatform.controller.sink.add(ConnectivityResult.none);
+      fakePlatform.controller.sink.add(ConnectivityResult.wifi);
+      fakePlatform.controller.sink.add(ConnectivityResult.none);
+      fakePlatform.controller.sink.add(ConnectivityResult.wifi);
+      fakePlatform.controller.sink.add(ConnectivityResult.none);
+      fakePlatform.controller.sink.add(ConnectivityResult.wifi);
+      fakePlatform.controller.sink.add(ConnectivityResult.none);
+      fakePlatform.controller.sink.add(ConnectivityResult.wifi);
+    });
+
+    test(
```

Comment:
Is this testing something different from the one in `graphql_test`?

Reply:
No, I duplicated here up a layer to ensure functionality in case another rewrite of the bloc happens.